File: 7/7.go
package main

import "math"

func reverse(x int) int {
	var y, rev int64 = int64(x), 0

	for y != 0 {
		rev = rev*10 + y%10

		// Check for overflow
		if rev > math.MaxInt32 || rev < math.MinInt32 {
			return 0
		}

		x /= 10
	}

	return int(rev)
}

File: 1-99/7/7.c
#include <stdio.h>
#include <limits.h>

int reverse(int x);

int main()
{
	printf("%d,%d\n", 123, reverse(123));
	printf("%d,%d\n", 1, reverse(1));
	printf("%d,%d\n", 100, reverse(100));
	printf("%d,%d\n", -123, reverse(-123));
	printf("%d,%d\n", 1534236469, reverse(1534236469));

	return 0;
}

int reverse(int x)
{
	long rev = 0;

	while (x != 0) {
		rev = rev * 10 + x % 10;

		// Check for overflow
		if (rev > INT_MAX || rev < INT_MIN)
			return 0;

		x /= 10;
	}

	return (int)rev;
}
